src/schema.py

```diff
@@ -95,15 +99,104 @@ class Meta:
         model = CapacityModel
 
 
+# MARK: - User
+
+
+class User(SQLAlchemyObjectType):
+    class Meta:
+        model = UserModel
+
+
+class UserInput(graphene.InputObjectType):
+    net_id = graphene.String(required=True)
+    giveaway_id = graphene.Int(required=True)
+
+
+# MARK: - Giveaway
+
+
+class Giveaway(SQLAlchemyObjectType):
+    class Meta:
+        model = GiveawayModel
+
+    user_ids = graphene.List(lambda: User)
+
+    def resolve_user_ids(self, info):
+        query = User.get_query(info=info).filter(UserModel.giveaway_id == self.id)
+        return query
+
+
+# MARK: - Giveaway
+
+
+class GiveawayInstance(SQLAlchemyObjectType):
+    class Meta:
+        model = GiveawayInstanceModel
+
+
 # MARK: - Query
 
 
 class Query(graphene.ObjectType):
     gyms = graphene.List(Gym)
+    users_by_giveawayid = graphene.List(User, id=graphene.Int())
 
     def resolve_gyms(self, info):
         query = Gym.get_query(info)
         return query.all()
 
 
-schema = graphene.Schema(query=Query)
+def resolve_users_by_giveaway_id(self, info, id):
+    query = User.get_query(info).filter(UserModel.giveaway_ids.any(GiveawayInstance.giveaway_id == id))
+    return query.all()
+
+
+# MARK: - Mutation
+
+
+class CreateUser(graphene.Mutation):
+    class Arguments:
+        net_id = graphene.String()
+
+    user = graphene.Field(User)
+
+    def mutate(root, info, net_id):
+        existing_user = db_session.query(UserModel).filter(UserModel.net_id == net_id).first()
+        if existing_user:
+            return CreateUser(user=existing_user)
+
+        new_user = UserModel(net_id=net_id, giveaway_ids=[])
+        db_session.add(new_user)
+        db_session.commit()
+        return CreateUser(user=new_user)
+
+
+class EnterGiveaway(graphene.Mutation):
+    class Arguments:
+        user_net_id = graphene.String(required=True)
+        giveaway_id = graphene.Int(required=True)
+
+    success = graphene.Boolean()
+    giveaway_instance = graphene.Field(GiveawayInstance)
+
+    def mutate(self, info, user_net_id, giveaway_id):
+        user = db_session.query(UserModel).filter_by(net_id=user_net_id).first()
+        if not user:
+            return EnterGiveaway(success=False, giveaway_instance=None)
+
+        giveaway = db_session.query(GiveawayModel).get(giveaway_id)
+        if not giveaway or any(instance.giveaway_id == giveaway_id for instance in user.giveaway_ids):
+            return EnterGiveaway(success=False, giveaway_instance=None)
+
+        giveaway_instance = GiveawayInstanceModel(user_id=user.id, giveaway_id=giveaway_id, numEntries=1)
+        db_session.add(giveaway_instance)
+        db_session.commit()
+        return EnterGiveaway(success=True, giveaway_instance=giveaway_instance)
+
```

Comment:
```suggestion


class CreateGiveaway(graphene.Mutation):
    class Arguments:
        name = graphene.String(required=True)

    giveaway = graphene.Field(Giveaway)

    def mutate(self, info, name):
        giveaway = GiveawayModel(name=name)
        db_session.add(giveaway)
        db_session.commit()
        return CreateGiveaway(giveaway=giveaway)
```
We should create a mutation to create a giveaway. This should also allow you to test without having to write a SQL query in the DB directly.